### ISSUE TITLE
BeforeAfterAll not working (or not working as expected)

### ISSUE THREAD

**Issue by jbjornson** [etorreborre/specs2#398]
We are writing some integration tests where we would like to set up some resources at the beginning, run some tests using those resources and release those resources at the end. The `BeforeAfterAll` trait seems like it would be the right tool to solve that problem.  Unfortunately it doesn't seem to work the way I would expect.

Here is a simplified example (using specs2 version 3.6.2):
```scala
import org.specs2.mutable.Specification
import org.specs2.specification.BeforeAfterAll

class SimpleBeforeAfterAllTest extends Specification with BeforeAfterAll {

  def beforeAll = println("beforeAll")
  def afterAll  = println("afterAll")

  "Arithmetic" should {
    "add two numbers" in {
      1 + 1 mustEqual 2
    }
    "add three numbers" in {
      1 + 1 + 1 mustEqual 3
    }
  }
}
```

It would expect this to work, but it results in the following output, which doesn't execute the tests or the `afterAll` code:
```
Testing started at 15:40 ...
beforeAll
Empty test suite.

Process finished with exit code 0
```

Is this a defect or am I simply doing something wrong?  Assuming that I'm doing something wrong, can you please provide an equally simple working example?

**Comment by etorreborre** [comment]
You need to tell me how you run your specification. If I run it with sbt I get
```
beforeAll
[info] TestSpec
[info]
[info] Arithmetic should
afterAll
[info]   + add two numbers
[info]   + add three numbers
[info]
[info] Total for specification TestSpec
[info] Finished in 49 ms
[info] 2 examples, 4 expectations, 0 failure, 0 error
```

**Comment by etorreborre** [comment]
If you are using sbt you might want to check the value of the `testFrameworks` settings, maybe specs2 is not registered? (possibly because you are using an `integration` config of some sort?)

**Comment by jbjornson** [comment]
Looking at your output, the order of execution is incorrect: `afterAll` is printed before the two tests are run.

We are running the tests with sbt (via IntelliJ), so I'll have a look at our setup and get back to you.  
I'll also run the tests directly in sbt to see if there is a difference.

**Comment by etorreborre** [comment]
The display of examples and their execution is asynchronous. If I run the following specification:
```scala

class TestSpec extends Specification with BeforeAfterAll {

  def beforeAll = println("beforeAll")
  def afterAll  = println("afterAll")

  "Arithmetic" should {
    "add two numbers" in {
      println("example1")
      1 + 1 mustEqual 2
    }
    "add three numbers" in {
      println("example2")
      1 + 1 + 1 mustEqual 3
    }
  }
}
```
 I get:
```scala
[info] Compiling 1 Scala source to /Users/etorreborre/projects/specs2/specs2-test/target/scala-2.11/test-classes...
beforeAll
example2
example1
[info] TestSpec
[info]
[info] Arithmetic should
afterAll
[info]   + add two numbers
[info]   + add three numbers
[info]
[info] Total for specification TestSpec
[info] Finished in 77 ms
[info] 2 examples, 4 expectations, 0 failure, 0 error
```
However I don't see `afterAll` being printed with Intellij which looks like a bug:
```
beforeAll
example1
example2
Empty test suite.
```

I'll have a look at that and come back to you.

**Comment by jbjornson** [comment]
I can confirm that it works command line but doesn't run the tests when executed in IntelliJ (but the `beforeAll`and `afterAll` actions are executed).

**Comment by etorreborre** [comment]
I added methods on the `Notifier` interface which broke the integration with Intellij. This is now fixed in `3.6.3-20150727123318-aea42fc`.
